### PR TITLE
Python package set major updates

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/shippai/default.nix [hunk 1]
```diff
@@ -3,11 +3,11 @@
 buildPythonPackage rec {
   pname = "shippai";
   # Please make sure that vdirsyncer still builds if you update this package.
-  version = "0.2.4";
```
Comment (gebner) [inline pkgs/development/python-modules/shippai/default.nix]: Does vdirsyncer still build?

File: pkgs/development/python-modules/sphinx/default.nix [hunk 2]
```diff
@@ -25,7 +25,7 @@
 }:
 
 buildPythonPackage rec {
-  pname = "sphinx";
+  pname = "Sphinx";
```
Comment (dotlambda) [inline pkgs/development/python-modules/sphinx/default.nix]: @Synthetica9 Was there a particular reason for the change? It causes sphinx to fail to build so I reverted it.

Comment (Synthetica9) [reply pkgs/development/python-modules/sphinx/default.nix]: See #50269

**Review by gebner — CHANGES_REQUESTED** [review]
I just tried to build `vdirsyncer` (now that rustc is in the cache), and it fails to build because of the `shippai` update.  Could you please revert the `shippai` update?

**Comment by dotlambda** [comment]
@ryantm Most of these conflicts are due to your bot upgrading python packages. I think you at least shouldn't do python upgrades when the package has already been upgraded in one of the staging/staging-next/python-unstable branches.

**Comment by FRidh** [comment]
@dotlambda it's something worth considering, but I am not sure whether it is worth spending time on. These kind of merge conflicts are at least trivial to resolve, though the hash does always look differently.

**Comment by dotlambda** [comment]
Alternatively, the update script could use long hashes for `fetchPypi`.

**Comment by ryantm** [comment]
@dotlambda I check master and staging currently. I'll add staging-next and python-unstable.

**Comment by FRidh** [comment]
@ryantm not python-unstable, there is no guarantee that changes in there go in.

**Comment by dotlambda** [comment]
@FRidh Do you have an idea what to do about the dateparser test failures?

**Comment by dotlambda** [comment]
@FRidh I think there are more failing packages on master than on python-unstable. What's holding us back from merging?

**Comment by dotlambda** [comment]
I just rebased on staging and believe we should merge if Hydra doesn't show any major breakage.

**Comment by dotlambda** [comment]
I just rebased on staging and believe we should merge if Hydra doesn't show any major breakage.

**Comment by Mic92** [comment]
When I have other minor python library updates that depend on updates here, should I just push them to the branch here as well?

**Comment by dotlambda** [comment]
> When I have other minor python library updates that depend on updates here, should I just push them to the branch here as well?

Sounds like a sensible thing to do. After you're done with that, we should *definitely* merge this PR!

**Comment by Mic92** [comment]
@dotlambda you go ahead with merging. I will add it later.

**Comment by dotlambda** [comment]
@Mic92 I prefer to let @FRidh do that.

**Comment by dotlambda** [comment]
@GrahamcOfBorg eval

**Pull request merged**